core/src/main/java/org/apache/gravitino/meta/UserEntity.java

```diff
@@ -44,18 +45,22 @@ public class UserEntity implements User, Entity, Auditable, HasIdentifier {
   public static final Field AUDIT_INFO =
       Field.required("audit_info", AuditInfo.class, "The audit details of the user entity.");
 
-  public static final Field ROLE_NAMES =
-      Field.optional("role_names", List.class, "The role names of the user entity");
+  public static final Field ROLE_NAMES_SUPPLIER =
+      Field.required(
+          "role_names_supplier", Supplier.class, "The role names supplier of the user entity");
 
-  public static final Field ROLE_IDS =
-      Field.optional("role_ids", List.class, "The role ids of the user entity");
+  public static final Field ROLE_IDS_SUPPLIER =
+      Field.required(
+          "role_ids_supplier", Supplier.class, "The role ids supplier of the user entity");
 
   private Long id;
   private String name;
   private AuditInfo auditInfo;
-  private List<String> roleNames;
-  private List<Long> roleIds;
   private Namespace namespace;
+  // The roleIds is a lazy field to avoid unnecessary cost.
+  private Supplier<List<Long>> roleIdsSupplier = () -> null;
+  // The roleNames is a lazy field to avoid unnecessary cost.
+  private Supplier<List<String>> roleNamesSupplier = () -> null;
```

Comment:
I think we need to use one `Role` object variable to replace `roleIds` and `roleNames` two variable.